libnmstate/validator.py

```diff
@@ -357,6 +359,40 @@ def _assert_vlan_filtering_trunk_tag(trunk_tag_state):
             )
 
 
+def _assert_vlan_mtu_not_greater_than_base_iface(
+    original_desired_state, current_state
+):
+    # validate if vlan mtu is greater than base interface mtu
+    for iface_name, iface_state in original_desired_state.interfaces.items():
+        if iface_state.get(Interface.TYPE) == InterfaceType.VLAN:
+            if iface_state.get(Interface.STATE) == InterfaceState.UP:
+                base_iface_name = iface_state.get(VLAN.CONFIG_SUBTREE).get(
+                    VLAN.BASE_IFACE
+                )
+                # check if base_iface is in desired state.
+                # if not then get from current state
+                base_iface_state = original_desired_state.interfaces.get(
+                    base_iface_name
+                )
+                if not base_iface_state:
+                    base_iface_state = current_state.interfaces.get(
+                        base_iface_name
+                    )
+                if not base_iface_state:
+                    raise ValueError(
+                        f"base interface {base_iface_name} for vlan \
+                        {iface_name} does not exist"
+                    )
```

Comment:
This if statement could be inside of the one above, this way it improves readability.